be/src/pipeline/pipeline_fragment_context.cpp

```diff
@@ -824,11 +750,23 @@ Status PipelineFragmentContext::_create_sink(int sender_id, const TDataSink& thr
 }
 
 void PipelineFragmentContext::_close_action() {
-    _runtime_profile->total_time_counter()->update(_fragment_watcher.elapsed_time());
-    send_report(true);
-    _stop_report_thread();
+    auto close_time = _fragment_watcher.elapsed_time();
+    _runtime_profile->total_time_counter()->update(close_time);
+    COUNTER_UPDATE(_close_timer, close_time);
     // all submitted tasks done
-    _exec_env->fragment_mgr()->remove_pipeline_context(shared_from_this());
+    _exec_env->send_report_thread_pool()->submit_func([&, this] {
```

Comment:
It is very dangerous because if close_action returns then other code may release the plan or clean object pool or delete some objects.
It will core in the async thread.

Reply:
`remove_pipeline_context` will exected after `send_report`. So, I think `PipelineFragmentContext` can not be released.